### PR TITLE
Skip Flaky Checks

### PR DESCRIPTION
**Pull request by foodprocessor**
### What type of Pull Request is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

### Describe your changes in brief
Don't run test that trips on LocalStack issue

### Checklist

- [ ] Tested locally
- [ ] Added new dependencies
- [ ] Updated documentation
- [ ] Added tests

### Related Issues

- Related Issue #
- Closes #

### CODE REVIEW

File: test/e2e_tests/dir_test.go [hunk 1]
```diff
@@ -527,7 +527,12 @@ func (suite *dirTestSuite) TestGitStash() {
 }
 
 func (suite *dirTestSuite) TestReadDirLink() {
-	if runtime.GOOS == "windows" || suite.adlsTest && strings.ToLower(enableSymlinkADLS) != "true" {
+	// Symbolic link creation requires admin rights on Windows.
```
Comment (foodprocessor) [inline test/e2e_tests/dir_test.go:530]: This doesn't change functionality, but is really a fixup for another change I made.
I had added the OS check to make one long conditional, even though it was meant for the ADLS flag.
So this change separates them, along with a better print.

File: component/s3storage/s3storage_test.go [hunk 2]
```diff
@@ -2087,6 +2087,16 @@ func (s *s3StorageTestSuite) TestReadDir() {
 
 func (s *s3StorageTestSuite) TestStreamDirSmallCountNoDuplicates() {
 	defer s.cleanupTest()
+
+	// due to LocalStack issue https://github.com/localstack/localstack/issues/9605
+	// LocalStack will list the subdirectory twice
+	// the only guess we have here for whether LocalStack is running is using the OS
+	//   since LocalStack only runs on Linux in our tests
+	if runtime.GOOS != "windows" {
```
Comment (jfantinhardesty) [inline component/s3storage/s3storage_test.go:2095]: With the bug fix to local stack, do we need this, or can we just make sure we are using the latest docker version of localstack?

**Review by jfantinhardesty — APPROVED** [review]

File: component/s3storage/s3storage_test.go (hunk 2 above)
Comment (foodprocessor) [reply component/s3storage/s3storage_test.go:2095]: Greate! I'll drop this!

**Review by Ferelith-maker — APPROVED** [review]

File: component/s3storage/s3storage_test.go [hunk 3]
```diff
@@ -2087,6 +2087,16 @@ func (s *s3StorageTestSuite) TestReadDir() {
...
+		fmt.Println("Skipping this test for LocalStack")
```
[the remark this answers is not in the document]
Comment (foodprocessor) [reply component/s3storage/s3storage_test.go]: This is a struggle with all these removed tests. A flaky test really needs to be debugged and fixed with a root cause. But the stability of our CI pipeline is more urgent.

File: test/e2e_tests/dir_test.go [hunk 4]
```diff
@@ -527,7 +527,12 @@ func (suite *dirTestSuite) TestGitStash() {
...
+	if runtime.GOOS == "windows" {
+		fmt.Println("Skipping test for Windows")
```
[the remark this answers is not in the document]
Comment (foodprocessor) [reply test/e2e_tests/dir_test.go:532]: Good point!

File: component/s3storage/s3storage_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (foodprocessor) [reply component/s3storage/s3storage_test.go]: We ought to put some TODOs above some of these, explaining which check failed and what we think might be done to fix it (if we know of anything). I'd have to go asking which tests are TODOs and which are just not possible (e.g. symlinks and special characters not supported on Windows), so I'd like to punt this for later.

**Review by jfantinhardesty — APPROVED** [review]
Looks good. Also seems that "TestOpenFileNotInCache" is flaky on Windows since it just failed. Could you skip that one on Windows too?

**Pull request merged**